experimental/dds/tree/src/EditLog.ts

```diff
@@ -258,30 +287,46 @@ export class EditLog<TChange = unknown> extends TypedEventEmitter<IEditLogEvents
 			this.registerEditAddedHandler(handler);
 		}
 
-		this.editChunks = new BTree<number, EditChunk<TChange>>(undefined, compareFiniteNumbers);
+        assert(numberOfAvailableSequencedEdits > 0 && Number.isInteger(numberOfAvailableSequencedEdits), 'The numberOfAvailableSequencedEdits param must be a positive integer');
+        this._numberOfAvailableSequencedEdits = numberOfAvailableSequencedEdits;
+        this._firstAvailableIndex = Math.max(0, editIds.length - numberOfAvailableSequencedEdits);
 
-		editChunks.forEach((editChunkOrHandle) => {
-			const { startRevision, chunk } = editChunkOrHandle;
+		this.editChunks = new BTree<number, EditChunk<TChange>>(undefined, compareFiniteNumbers);
+        let i = editChunks.length - 1;
+		do {
+            const { startRevision, chunk } = editChunks[i];
 
 			if (isEditHandle(chunk)) {
 				this.editChunks.set(startRevision, { handle: chunk });
 			} else {
 				this.editChunks.set(startRevision, { edits: chunk as EditWithoutId<TChange>[] });
 			}
-		});
+        } while (editChunks[i--].startRevision > this._firstAvailableIndex && i >= 0);
 
-		this.sequencedEditIds = editIds.slice();
+		this.sequencedEditIds = new Denque(editIds.slice(this.firstAvailableIndex));
 
-		this.indexOfFirstEditInSession = indexOfFirstEditInSession;
-		this.maximumEvictableIndex = this.indexOfFirstEditInSession - 1;
+		this._indexOfFirstEditInSession = indexOfFirstEditInSession;
 
-		this.sequencedEditIds.forEach((id, index) => {
-			const encounteredEditId = this.allEditIds.get(id);
+        for (let i = 0; i < this.sequencedEditIds.length; i++) {
+            const id = assertNotUndefined(this.sequencedEditIds.get(i));
+            const encounteredEditId = this.allEditIds.get(id);
 			assert(encounteredEditId === undefined, 'Duplicate acked edit.');
-			this.allEditIds.set(id, { isLocal: false, index });
+			this.allEditIds.set(id, { isLocal: false, index: i + this.firstAvailableIndex });
+        }
+
+        this.loadedChunkCache = new LRU({
+			max: loadedChunkCacheSize,
+			noDisposeOnSet: true,
+			dispose: (revision, chunk) => {
+				chunk.edits = undefined;
+			},
 		});
 	}
 
+    private isValidIndex(index: number): boolean {
+        return Number.isInteger(index) && index >= this.firstAvailableIndex && index < this.length;
```

Comment:
I think it's fine to drop the integer validation here, especially for a private method. There are lots of places in SharedTree where we don't verify this, so let's stick with precedent for now. Just make sure it's documented that it must be an integer.